### PR TITLE
Adding a cacheTime config option for assets

### PR DESCRIPTION
**Pull request by patrickconroy**
Right now all assets are cached for 1 day, which should probably be configurable.

Usage in bootstrap..

```
DispatcherFactory::add('Asset', ['cacheTime' => '+1 hour']);
```

If this needs a test let me know and I'll write one.

### CODE REVIEW

File: src/Routing/Filter/AssetFilter.php [hunk 1]
```diff
@@ -38,6 +38,27 @@ class AssetFilter extends DispatcherFilter
...
+     * Constructor.
+     *
+     * @param array $config Array of config.
+     */
+    public function __construct($config = [])
+    {
+        if (!empty($config['cacheTime'])) {
+            $this->_cacheTime = $config['cacheTime'];
+        }
+        parent::__construct($config);
```
Comment (markstory) [inline src/Routing/Filter/AssetFilter.php]: Would it be good to support strtotime() strings as you have or callables?

File: src/Routing/Filter/AssetFilter.php [hunk 2]
```diff
@@ -38,6 +38,27 @@ class AssetFilter extends DispatcherFilter
     protected $_priority = 9;
 
     /**
+     * The amount of time to cache the asset.
+     *
+     * @var string
+     */
+    protected $_cacheTime = '+1 day';
```
Comment (dereuromark) [inline src/Routing/Filter/AssetFilter.php]: I wonder if it should be "cacheDuration" or sth instead of "time"?

Comment (ADmad) [inline src/Routing/Filter/AssetFilter.php]: Yup, `cacheDuration` would be more appropriate.

Comment (ravage84) [inline src/Routing/Filter/AssetFilter.php]: `cacheDuration` :+1:

**Comment by lorenzo** [comment]
I would rather remove the cache than adding more options to this. The recommendation is that people use the symlinks instead of this filter.

**Comment by patrickconroy** [comment]
I'd be fine with removing it completely.. what's the consensus here? I can update this PR or make a new one just let me know.

**Comment by patrickconroy** [comment]
Any update here? No cache, or making it settable?

**Comment by markstory** [comment]
While I agree that people should use symlinks in prod, having the AssetFilter is pretty handy in development. This makes an existing default configurable, which gets a :+1: from me.

**Comment by lorenzo** [comment]
@patrickconroy Can you add this to the docs?

**Comment by dereuromark** [comment]
My remark regarding the better naming hasnt been addressed yet.

**Comment by patrickconroy** [comment]
@dereuromark Sorry I can change that now, I thought I already did it. Also, I went with `time` since that's how the `Cake\Network\Response::cache` method names its second parameter, which is what this is.
Stupid question, since this was just merged should I make a new PR with that change?
@lorenzo Sure, I'll do that now.

**Comment by dereuromark** [comment]
The difference with the response object is that there it is used inside strtotime(), so it does not have to be a "duration". Like "when do you want to invalidate?" Not sure what applies here, as we add a duration length on top from the looks.

**Pull request merged**